src/util.ts

```diff
@@ -2785,6 +2785,23 @@ export class Util {
         }
         return resultType;
     }
+
+    public findCommonObjects<T>(...arrays: T[][]): T[] {
```

Comment:
Might be something to address later, but array.includes will basically do a sequential loop over every element in the array. Could be pretty expensive. Is there any chance we could change the storage of these items into `Set` instead? Then the .has will be a hash lookup (much faster). But adding to those sets is slightly slower than arrays. 

I'm not sure the context here, just a thought.

Reply:
Changed to finding Set intersections...